runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -243,19 +252,53 @@ function TSHighlighter:get_query(lang)
   return self._queries[lang]
 end
 
+--- @param match table<integer,TSNode[]>
+--- @param bufnr integer
+--- @param capture integer
+--- @param metadata vim.treesitter.query.TSMetadata
+--- @return string?
+local function get_url(match, bufnr, capture, metadata)
+  ---@type string|number|nil
+  local url = metadata[capture] and metadata[capture].url
+
+  if type(url) ~= 'number' then
+    return
```

Comment:
`url` _could_ be a string, though it's likely not a common use case. Example:

```query
(#set! @foo "url" "https://example.com")
```

Reply:
I haven't meaningfully changed the code here. I've just adapted it to the return-early pattern. Do you want me to remove this term?

Reply:
Oh wait, yes I have good catch, I'll fix.